### DISCUSSION TITLE
Mehr Werte einpflegen

### DISCUSSION THREAD

**Discussion by tommyl2210** [gickowtf/enpal-homeassistant#7] [Ideas]
@gickowtf Vielen Dank für die Integration der Enpal 2 Box! MIttlerweile gibt es ja noch eine ganze Reihe mehr Werte, die auslesbar sind. Auf welchem Weg kann ich Dir am besten die Erweiterung zukommen lassen? Ein Fork auf dieses repository geht zurück auf das orignal repository.

**Comment by gickowtf** [comment]
Kannst du gerne hier als Code hinterlassen 😉

**Comment by tommyl2210** [comment]
Hi, hier die erweiterte sensor.py. Es gibt natürlich noch viel mehr Werte, aber mit denen kann ich gut leben :)

        #inverter
        if measurement == "inverter" and field == "Power.DC.Total":
            to_add.append(EnpalSensor(field, measurement, 'mdi:solar-power', 'Enpal Solar Production Power', config['enpal_host_ip'], config['enpal_host_port'], config['enpal_token'], 'power', 'W'))
        if measurement == "inverter" and field == "Power.House.Total":
            to_add.append(EnpalSensor(field, measurement, 'mdi:home-lightning-bolt', 'Enpal Power House Total', config['enpal_host_ip'], config['enpal_host_port'], config['enpal_token'], 'power', 'W'))
        if measurement == "inverter" and field == "Voltage.String.1":
            to_add.append(EnpalSensor(field, measurement, 'mdi:home-lightning-bolt', 'Enpal Voltage String 1', config['enpal_host_ip'], config['enpal_host_port'], config['enpal_token'], 'voltage', 'V'))
        if measurement == "inverter" and field == "Voltage.String.2":
            to_add.append(EnpalSensor(field, measurement, 'mdi:home-lightning-bolt', 'Enpal Voltage String 2', config['enpal_host_ip'], config['enpal_host_port'], config['enpal_token'], 'voltage', 'V'))
        if measurement == "inverter" and field == "Current.String.1":
            to_add.append(EnpalSensor(field, measurement, 'mdi:home-lightning-bolt', 'Enpal Current String 1', config['enpal_host_ip'], config['enpal_host_port'], config['enpal_token'], 'current', 'A'))
        if measurement == "inverter" and field == "Current.String.2":
            to_add.append(EnpalSensor(field, measurement, 'mdi:home-lightning-bolt', 'Enpal Current String 2', config['enpal_host_ip'], config['enpal_host_port'], config['enpal_token'], 'current', 'A'))
        if measurement == "inverter" and field == "Frequency.Grid":
            to_add.append(EnpalSensor(field, measurement, 'mdi:sine-wave', 'Enpal Frequency Grid', config['enpal_host_ip'], config['enpal_host_port'], config['enpal_token'], 'frequency', 'Hz'))
        if measurement == "inverter" and field == "Temperature.Housing.Inside":
            to_add.append(EnpalSensor(field, measurement, 'mdi:thermometer', 'Enpal Temperature Housing Inside', config['enpal_host_ip'], config['enpal_host_port'], config['enpal_token'], 'temperature', '°C'))

        #systen
        if measurement == "system" and field == "Power.External.Total":
            to_add.append(EnpalSensor(field, measurement, 'mdi:home-lightning-bolt', 'Enpal Power External Total', config['enpal_host_ip'], config['enpal_host_port'], config['enpal_token'], 'power', 'W'))
        # Consum Total per Day
        if measurement == "system" and field == "Energy.Consumption.Total.Day":
            to_add.append(EnpalSensor(field, measurement, 'mdi:home-lightning-bolt', 'Enpal Energy Consumption', config['enpal_host_ip'], config['enpal_host_port'], config['enpal_token'], 'energy', 'kWh'))

        # to the Grid and from the Grid
        if measurement == "system" and field == "Energy.External.Total.Out.Day":
            to_add.append(EnpalSensor(field, measurement, 'mdi:transmission-tower-export', 'Enpal Energy External Out Day', config['enpal_host_ip'], config['enpal_host_port'], config['enpal_token'], 'energy', 'kWh'))
        if measurement == "system" and field == "Energy.External.Total.In.Day":
            to_add.append(EnpalSensor(field, measurement, 'mdi:transmission-tower-import', 'Enpal Energy External In Day', config['enpal_host_ip'], config['enpal_host_port'], config['enpal_token'], 'energy', 'kWh'))

        # Solar Energy.Production.Total.Day
        if measurement == "system" and field == "Energy.Production.Total.Day":
            to_add.append(EnpalSensor(field, measurement, 'mdi:solar-power-variant', 'Enpal Production Day', config['enpal_host_ip'], config['enpal_host_port'], config['enpal_token'], 'energy', 'kWh'))
            
        #Power Sensor
        if measurement == "powerSensor" and field == "Voltage.Phase.A":
            to_add.append(EnpalSensor(field, measurement, 'mdi:lightning-bolt', 'Enpal Voltage Phase A', config['enpal_host_ip'], config['enpal_host_port'], config['enpal_token'], 'voltage', 'V'))
        if measurement == "powerSensor" and field == "Current.Phase.A":
            to_add.append(EnpalSensor(field, measurement, 'mdi:lightning-bolt', 'Enpal Ampere Phase A', config['enpal_host_ip'], config['enpal_host_port'], config['enpal_token'], 'current', 'A'))
        if measurement == "powerSensor" and field == "Power.AC.Phase.A":
            to_add.append(EnpalSensor(field, measurement, 'mdi:lightning-bolt', 'Enpal Power Phase A', config['enpal_host_ip'], config['enpal_host_port'], config['enpal_token'], 'power', 'W'))
        if measurement == "powerSensor" and field == "Voltage.Phase.B":
            to_add.append(EnpalSensor(field, measurement, 'mdi:lightning-bolt', 'Enpal Voltage Phase B', config['enpal_host_ip'], config['enpal_host_port'], config['enpal_token'], 'voltage', 'V'))
        if measurement == "powerSensor" and field == "Current.Phase.B":
            to_add.append(EnpalSensor(field, measurement, 'mdi:lightning-bolt', 'Enpal Ampere Phase B', config['enpal_host_ip'], config['enpal_host_port'], config['enpal_token'], 'current', 'A'))
        if measurement == "powerSensor" and field == "Power.AC.Phase.B":
            to_add.append(EnpalSensor(field, measurement, 'mdi:lightning-bolt', 'Enpal Power Phase B', config['enpal_host_ip'], config['enpal_host_port'], config['enpal_token'], 'power', 'W'))        
        if measurement == "powerSensor" and field == "Voltage.Phase.C":
            to_add.append(EnpalSensor(field, measurement, 'mdi:lightning-bolt', 'Enpal Voltage Phase C', config['enpal_host_ip'], config['enpal_host_port'], config['enpal_token'], 'voltage', 'V'))
        if measurement == "powerSensor" and field == "Current.Phase.C":
            to_add.append(EnpalSensor(field, measurement, 'mdi:lightning-bolt', 'Enpal Ampere Phase C', config['enpal_host_ip'], config['enpal_host_port'], config['enpal_token'], 'current', 'A'))
        if measurement == "powerSensor" and field == "Power.AC.Phase.C":
            to_add.append(EnpalSensor(field, measurement, 'mdi:lightning-bolt', 'Enpal Power Phase C', config['enpal_host_ip'], config['enpal_host_port'], config['enpal_token'], 'power', 'W'))
        
        #Battery    
        if measurement == "battery" and field == "Power.Battery.Charge.Discharge":
            to_add.append(EnpalSensor(field, measurement, 'mdi:battery-charging', 'Enpal Battery Power', config['enpal_host_ip'], config['enpal_host_port'], config['enpal_token'], 'power', 'W'))
        if measurement == "battery" and field == "Energy.Battery.Charge.Level":
            to_add.append(EnpalSensor(field, measurement, 'mdi:battery', 'Enpal Battery Percent', config['enpal_host_ip'], config['enpal_host_port'], config['enpal_token'], 'battery', '%'))
        if measurement == "battery" and field == "Energy.Battery.Charge.Day":
            to_add.append(EnpalSensor(field, measurement, 'mdi:battery-arrow-up', 'Enpal Battery Charge Day', config['enpal_host_ip'], config['enpal_host_port'], config['enpal_token'], 'energy', 'kWh'))
        if measurement == "battery" and field == "Energy.Battery.Discharge.Day":
            to_add.append(EnpalSensor(field, measurement, 'mdi:battery-arrow-down', 'Enpal Battery Discharge Day', config['enpal_host_ip'], config['enpal_host_port'], config['enpal_token'], 'energy', 'kWh'))
        if measurement == "battery" and field == "Energy.Battery.Charge.Total.Unit.1":
            to_add.append(EnpalSensor(field, measurement, 'mdi:battery-arrow-up', 'Enpal Battery Charge Total', config['enpal_host_ip'], config['enpal_host_port'], config['enpal_token'], 'energy', 'kWh'))
        if measurement == "battery" and field == "Energy.Battery.Discharge.Total.Unit.1":
            to_add.append(EnpalSensor(field, measurement, 'mdi:battery-arrow-down', 'Enpal Battery Discharge Total', config['enpal_host_ip'], config['enpal_host_port'], config['enpal_token'], 'energy', 'kWh'))
        if measurement == "battery" and field == "Temperature.Battery.Unit.1":
            to_add.append(EnpalSensor(field, measurement, 'mdi:thermometer', 'Enpal Temperature Battery Unit 1', config['enpal_host_ip'], config['enpal_host_port'], config['enpal_token'], 'temperature', '°C'))


        #Wallbox
        if measurement == "wallbox" and field == "State.Wallbox.Connector.1.Charge":
            to_add.append(EnpalSensor(field, measurement, 'mdi:ev-station', 'Wallbox Charge Percent', config['enpal_host_ip'], config['enpal_host_port'], config['enpal_token'], 'battery', '%'))
        if measurement == "wallbox" and field == "Power.Wallbox.Connector.1.Charging":
            to_add.append(EnpalSensor(field, measurement, 'mdi:ev-station', 'Wallbox Charging Power', config['enpal_host_ip'], config['enpal_host_port'], config['enpal_token'], 'power', 'W'))
        if measurement == "wallbox" and field == "Energy.Wallbox.Connector.1.Charged.Total":
            to_add.append(EnpalSensor(field, measurement, 'mdi:ev-station', 'Wallbox Charging Total', config['enpal_host_ip'], config['enpal_host_port'], config['enpal_token'], 'energy', 'Wh'))

**Comment by sceife** [comment]
Vielen Dank für die tolle Integration. Lediglich mehr Werte würde ich mir auch wünschen.

Sonst ein dickes Lob.

**Comment by holoman5** [comment]
@ sceife
Ich habe die sensor.py an die Tabellennamen in der  Flux Datenbank angepasst und habe jetzt 17 Entitäten.
Werde später manuell noch Werte hinzufügen.
